splunk_add_on_ucc_framework/commands/build.py

```diff
@@ -321,6 +322,19 @@ def generate(
 ):
     logger.info(f"ucc-gen version {__version__} is used")
     logger.info(f"Python binary name to use: {python_binary_name}")
+
+    try:
+        python_binary_version = subprocess.run(
+            [python_binary_name, "--version"], stdout=subprocess.PIPE
+        ).stdout.decode("utf-8")
+    except subprocess.CalledProcessError as e:
+        logger.error(
+            f"Failed to identify Python version for library installation. Error: {e}"
+        )
+        sys.exit(1)
```

Comment:
Can you please move to separate function? It can be in this file. And please add a unit test for that as well.

Reply:
added unit tests and moved identification logic into separate function.